plugins/node/opentelemetry-plugin-ioredis/src/utils.ts

```diff
@@ -109,7 +109,20 @@ export const traceSendCommand = (
 
     try {
       const result = original.apply(this, arguments);
-      endSpan(span, null);
+
+      const origResolve = cmd.resolve;
+      /* eslint-disable @typescript-eslint/no-explicit-any */
+      cmd.resolve = function (result: any) {
+        endSpan(span, null);
+        origResolve(result);
+      };
+
+      const origReject = cmd.reject;
+      cmd.reject = function (err: Error) {
+        endSpan(span, err);
```

Comment:
Ideally we should use `recordException` inside endSpan function as `err` might contain information also about stack and code but I see there is used `setStatus`

Reply:
I think it's not in the scope of this PR. I used the already existing function as it is.
Maybe you can open an issue for that?

Reply:
the description says `fix: set span status to error if server response with error. ` isn't this exactly connected with saving information about error ?

Reply:
This PR is about setting the error according to the server response, instead of falsely setting it according to function call returned value. I prefer to keep it minimal and to fix a single issue.
I agree that there is room for other improvements to this plugin code as you mentioned.

Reply:
Hi @obecny ,
Just checking about the status of this comment.
This PR is waiting for almost 3 months

Reply:
@obecny 
Do we agree that there is currently a bug described above in this plugin code?
Do we agree that the PR will not introduce **new** bugs or issues to the code according to your review?
Do we agree that the change in this PR is fixing the above issue?

If so, I have to say that I don't understand why you block it from being merged.
Is it the official policy of this project?

Reply:
@blumamir two things

1. technical - `recordException` functionality is part of official spec and is implemented in js. If error occurs we should store the information about error so that it is easy to understand what has happened. Stack trace is one of the element that helps debugging and fixing stuff. This will definitely help user to understand what has happened. For this purpose `recordException` has been added and I would say it is highly recommended when you store the information that the span has error.
 
2. Blaming me for blocking this PR and that the ticket has been opened so long. 
If you look look closely what are the rules for merging you will see that even if I wasn't here this PR could be merged if it meets the merging criteria which is doable even without my approve. You can also see that I already said in December that this can be merged just fix the lint errors otherwise I cannot merge or any1 else when the build is broken. It wasn't fixed for quite long time was just there (or correct me if I'm wrong but then nothing was said to just attract some attention for more than month) until @vmarchaud asked you to fix the build again. Then last week I was asked again for one more review which I did and raised the issue that it would make sense to use also `recordException` (btw 30s job, one line of code - it will be longer to create a new issue then adding it here). But you didn't like it not sure why though. 

If I wanted to block this particular PR I would click "request changes" then it would be blocked. But I didn't do that, you just don't have my approve as I think it still lacks `recordException`. You could raise a separate issue for that so it won't get lost and any1 else can pick it up and add it in separate PR. But I don't understand why I have to approve something if I think it still lacks things (in my human opinion). 

Your last sentence is then the least understandable for when you say about policy of this project, but I'm not going to comment on that. 

And as last thing, this is quite a big project some people that were approvers are not longer active here. There are more work and less people. We have recently decided it to improve that too. Have a new maintainer and 2 new approvers - how it will work we will see in next couple weeks, but I think it is already a big improvement.

Reply:
Thanks for the detailed response.
I appreciate the work and time you put into this project and didn't mean to blame you for anything. Sorry if it sounds like this. I thought your review blocks this PR from being merged, but I see now that it is not. I'll just wait for other reviews in that case.
I understand what you are saying about `recordException` and would love to see it get fixed as well, just don't want to mix it into this PR which is fixing something else.

Reply:
My understanding is that this unresolved conversation indeed blocks merge (along with the lack of 2 maintainer approvals)  as per https://github.com/open-telemetry/opentelemetry-js/blob/main/CONTRIBUTING.md#general-merge-requirements

> No unresolved conversations.

Which in itself is a questionable/hard to interpret requirement as the PR author could resolve the conversation themself.

Reply:
> My understanding is that this unresolved conversation indeed blocks merge (along with the lack of 2 maintainer approvals) as per https://github.com/open-telemetry/opentelemetry-js/blob/main/CONTRIBUTING.md#general-merge-requirements

@naseemkullah You are right but until the PR receives enough approves it doesn't block anything. Once it receives all needed approves then all unresolved conversations should be resolved so that nothing hangs behind. It still didn't happen after so long time. As I mentioned before any of following 2 will work for me: either creating a new issue so it is not forgotten and can be done in separate PR or add recordException. Both ways works fine for me. But it doesn't stop other people from reviewing it, the more reviews the better.

Reply:
@obecny I'll add the `recordException` tomorrow and update the PR.

Reply:
@obecny 
Added the `recordException` to the PR and updated the tests accordingly.